packages/wallet/src/wallet.ts

```diff
@@ -317,46 +395,39 @@ export class DemexWallet {
     }
   }
 
-  private async reloadAccountInfo() {
-    // carbon account always takes priority
-    let account: Account | undefined = undefined;
-    if (this.bech32Address)
-      account = await this.getAccount(this.bech32Address);
-    if (!account && this.evmBech32Address)
-      account = await this.getAccount(this.evmBech32Address);
-    return account
+  private async reloadAccountInfo(reloadAddresses: ReloadAddresses) {
+    const { address, evmBech32Address } = reloadAddresses
+    const account: Account | undefined = await this.getAccount(address);
+    if (account) return account
+    if (evmBech32Address) {
+      const evmAccount: Account | undefined = await this.getAccount(evmBech32Address);
+      if (evmAccount) return evmAccount
+    }
```

Comment:
recommend passing in the key used to index `this.accountStates` instead of custom `ReloadAddresses` type.